zend_abstract_interface/symbols/lookup.c

```diff
@@ -192,28 +165,26 @@ static inline zend_function *zai_symbol_lookup_function_impl(zai_symbol_scope_t
         key.ptr == name->ptr ? zai_symbol_lookup_clean(key) : key,
         scope_type != ZAI_SYMBOL_SCOPE_NAMESPACE, true);
 
-    if (key.ptr != name->ptr) {
-        pefree((char*) key.ptr, 1);
-    }
+    zai_string_dtor(&tmpkey);
 
     return function;
 }
 
 static inline zval* lookup_constant_global(zai_symbol_scope_t scope_type, void *scope, zai_str *name) {
     zai_str key = *name;
+    zai_string tmpkey = ZAI_STRING_EMPTY;
 
     if (scope_type == ZAI_SYMBOL_SCOPE_NAMESPACE) {
-        key = zai_symbol_lookup_key(scope, name, false);
+        tmpkey = zai_symbol_lookup_key(scope, name, false);
+        key = zai_string_as_str(&tmpkey);
     }
 
     zend_constant *constant = zai_symbol_lookup_table(
         EG(zend_constants),
         key.ptr == name->ptr ? zai_symbol_lookup_clean(key) : key,
         false, true);
 
-    if (key.ptr != name->ptr) {
-        pefree((char*) key.ptr, 1);
-    }
+    zai_string_dtor(&tmpkey);
```

Comment:
```suggestion
    if (scope_type == ZAI_SYMBOL_SCOPE_NAMESPACE) {
        zai_string_dtor(&tmpkey);
    }
```
and avoid initializing tmpkey.

Reply:
What's your thinking? It seems trickier that way, and I can't tell what gain it's supposed to have.